group1/communication/node/main.js

```diff
@@ -29,27 +29,46 @@ var done = false;
 app.use(multer({
         'dest': '../../comm/static/uploads',
         onFileUploadStart: function(file) {
+            if(testNameValidation(file.originalname)){
                 console.log(file.originalname + ' is starting...');
+            } else {
+                console.log('File name is invalid');
+                return false;
+            }
         },
         onFileUploadComplete: function(file) {
                 console.log(file.fieldname + ' uploaded to ' + file.path);
                 done = true;
         }
 }));
 
+function testNameValidation(text) {
+    var format = /[ !@#$%^&*()+\-=\[\]{};':"\\|,<>\/?]/;
+    if(text.trim() == null || text.trim() == "" || format.test(text) == true) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
 app.post('/upload', function(req,res){
         if (done==true) {
                 var hash = crypto.randomBytes(20).toString('hex');
-				var directory =  path.join(homedir + '/sites/www.3blueprints.com/static/uploads/');
+                console.log(path);
```

Comment:
Is there any purpose to logging the path to the console other than testing?  If there is another purpose, and it should be checked into source control, more information can be added to this statement for context and clarity.  Otherwise this can probably be removed.